### PR TITLE
[SAM] Initial 7.0 SAM Support

### PR DESCRIPTION
**Pull request by Rjb99**
This PR does the following:

Removes 6.x ghosts.
Adds 7.0 Actions and statuses
Converts Ogi to the CoreProc Module, adds the rest of the Ready Actions.
Updates Meikyo to also track Gyofu, the new hakaze update

### CODE REVIEW

File: src/data/ACTIONS/root/SAM.ts [hunk 1]
```diff
@@ -25,14 +19,28 @@ export const SAM = ensureActions({
 		},
 	},
 
+	GYOFU: {
+		id: 36963,
+		name: 'Gyofu',
+		icon: iconUrl(36963),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: The icon number for this is 3191, going off the asset URL for the icon on https://arcanedisgea.com/xiv/xiv-id/?index=Action&query=Gyofu

File: src/data/ACTIONS/root/SAM.ts [hunk 2]
```diff
@@ -124,23 +135,32 @@ export const SAM = ensureActions({
 		castTime: 1300,
 	},
 
+	TENDO_GOKEN: { //TODO: Verify Icon
+		id: 36965,
+		name: 'Tendo Goken',
+		icon: iconUrl(36965),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: Icon # 3193 :)

File: src/data/ACTIONS/root/SAM.ts [hunk 3]
```diff
@@ -171,6 +191,16 @@ export const SAM = ensureActions({
 		castTime: 1300,
 	},
 
+	TENDO_SETSUGEKKA: { //TODO: Verify, Icon
+		id: 36966,
+		name: 'Tendo Setsugekka',
+		icon: iconUrl(36966),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3194

File: src/data/ACTIONS/root/SAM.ts [hunk 4]
```diff
@@ -194,6 +224,50 @@ export const SAM = ensureActions({
 		},
 	},
 
+	TSUBAME_GAESHI: {
+		id: 16483,
+		name: 'Tsubame Gaeshi',
+		icon: iconUrl(16483),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3180

File: src/data/ACTIONS/root/SAM.ts [hunk 5]
```diff
@@ -194,6 +224,50 @@ export const SAM = ensureActions({
...
+		onGcd: true,
+		speedAttribute: Attribute.SKILL_SPEED,
+	},
+
+	KAESHI_GOKEN: {
+		id: 16485,
+		name: 'Kaeshi: Goken',
+		icon: iconUrl(16485),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3182

File: src/data/ACTIONS/root/SAM.ts [hunk 6]
```diff
@@ -194,6 +224,50 @@ export const SAM = ensureActions({
...
+		onGcd: true,
+		speedAttribute: Attribute.SKILL_SPEED,
+		gcdRecast: 2500,
+	},
+
+	KAESHI_SETSUGEKKA: {
+		id: 16486,
+		name: 'Kaeshi: Setsugekka',
+		icon: iconUrl(16486),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3183

File: src/data/ACTIONS/root/SAM.ts [hunk 7]
```diff
@@ -194,6 +224,50 @@ export const SAM = ensureActions({
...
+		onGcd: true,
+		speedAttribute: Attribute.SKILL_SPEED,
+		gcdRecast: 2500,
+	},
+
+	TENDO_KAESHI_GOKEN: { //TODO Id, Icon, Recast
+		id: 36967,
+		name: 'Tendo Kaeshi Goken',
+		icon: iconUrl(36967),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3195

File: src/data/ACTIONS/root/SAM.ts [hunk 8]
```diff
@@ -194,6 +224,50 @@ export const SAM = ensureActions({
...
+		onGcd: true,
+		speedAttribute: Attribute.SKILL_SPEED,
+		gcdRecast: 2500,
+	},
+
+	TENDO_KAESHI_SETSUGEKKA: { //TODO Id, Icon, Recast
+		id: 36968,
+		name: 'Tendo Kaeshi Setsugekka',
+		icon: 'icon',
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3196

File: src/data/ACTIONS/root/SAM.ts [hunk 9]
```diff
@@ -222,13 +296,23 @@ export const SAM = ensureActions({
 		cooldown: 15000,
 		statusesApplied: ['THIRD_EYE'],
 	},
+	TENGETSU: {
+		id: 36962,
+		name: 'Tengetsu',
+		icon: iconUrl(36962),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3190

File: src/data/ACTIONS/root/SAM.ts [hunk 10]
```diff
@@ -357,10 +397,10 @@ export const SAM = ensureActions({
...
 
-	SHOHA_II: { //WHY SE WHY
-		id: 25779,
-		name: 'Shoha II',
-		icon: iconUrl(3185),
-		cooldown: 15000,
+	ZANSHIN: {
+		id: 36964,
+		name: 'Zanshin',
+		icon: iconUrl(36964),
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts]: 3192

**Review by AkairyuGestalter — CHANGES_REQUESTED** [review]
Overall I think these changes seem sane. Left comments with the correct icon numbers for the new actions to save you needing to look them up!

**Comment by Rjb99** [comment]
icon fixes were made, sorry for the trouble. idk what my brain was doing when I entered those numbers.

File: src/data/ACTIONS/root/SAM.ts [hunk 11]
```diff
@@ -25,14 +19,28 @@ export const SAM = ensureActions({
 		},
 	},
 
+	GYOFU: {
+		id: 36963,
```
Comment (AkairyuGestalter) [inline src/data/ACTIONS/root/SAM.ts:23]: Need to add this to the combo steps for Yukikaze too, it's throwing tons of broken combo issues

File: src/parser/jobs/sam/modules/ReadyProcs.tsx [hunk 12]
```diff
@@ -0,0 +1,120 @@
...
+		},
+		{
+			procStatus: this.data.statuses.TENDO,
+			consumeActions: [
+				this.data.actions.TENDO_GOKEN,
+				this.data.actions.TENDO_SETSUGEKKA,
+			],
+		},
+		{
+			procStatus: this.data.statuses.TSUBAME_GAESHI_READY,
```
Comment (AkairyuGestalter) [inline src/parser/jobs/sam/modules/ReadyProcs.tsx]: Something is going on with this, might be a goof with the status data?
image

**Review by AkairyuGestalter — CHANGES_REQUESTED** [review]
Two notes I saw while poking at some logs. Should we also consider updating the links in the About and Sen table header on this PR or is that still being workshopped and should wait till a follow-up?

File: src/parser/jobs/sam/modules/ReadyProcs.tsx (hunk 12 above)
Comment (Rjb99) [reply src/parser/jobs/sam/modules/ReadyProcs.tsx]: fixed typo.

**Comment by Rjb99** [comment]
This should be ready to land now.

**Comment by Rjb99** [comment]
I hope I don't accidently revert this again...

image
image

Hagakure has been knocked to a minor suggest as a 2.17 full uptime loop that requires no filler has been found.

**Review by AkairyuGestalter — APPROVED** [review]
Looks good now!

**Pull request merged**